From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 13 Nov 2019 13:38:20 -0500
Subject: [PATCH] gitignore: add our build directories.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 .gitignore | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/.gitignore b/.gitignore
index 970b638e0c0..014f7d02f46 100644
--- a/.gitignore
+++ b/.gitignore
@@ -4,3 +4,9 @@
 *.a
 *.tar.*
 *.tar
+aa64
+arm
+ia32
+ia64
+mips64el
+x64
